### ISSUE TITLE
$ref in swagger-editor running locally 

### ISSUE THREAD

**Issue by webron** [swagger-api/swagger-editor#639]
_From @Tasneemb on September 10, 2015 8:1_

Hi 

I am running a local instance of swagger-editor. I had a single yaml file which had no errors in it. I then decided to split the files for each of the paths defined in the single yaml file. Since splitting , I am getting this error in my validator window...

Additional properties not allowed: $ref
Relative paths to the individual endpoints. They must be relative to the 'basePath'.
Jump to line 21
Details
Object
code: "OBJECT_ADDITIONAL_PROPERTIES"
message: "Additional properties not allowed: $ref"
path: Array [1]
description: "Relative paths to the individual endpoints. They must be relative to the 'basePath'."
level: 900

this is the part which is in error
paths:
  $ref: '#/paths/index.yaml'

Please advise....

_Copied from original issue: swagger-api/swagger-parser#88_

**Comment by mohsen1** [comment]
Can you write up a minimal swagger that produces this error?

**Comment by Tasneem1434** [comment]
Hi. 
I have the index.html
swagger: '2.0'
info:
  version: 1.0.0
  title: My Test API
basePath: /1.0.0
schemes:
  - http
  - https
consumes:
  - application/json
produces:
  - application/json
  # - application/xml
tags:
  - name: core
    description: Core functionality of the API
  - name: audit
    description: |
      Information about how the API is used
paths:
  $ref: ./paths/index.yaml
securityDefinitions:
  $ref: ./securityDefinitions/index.yaml  
definitions:
  $ref: ./definitions/index.yaml    

then I have paths folder with index.yaml and car.yaml
index.yaml
/car/{carId}/{companyId}:
  $ref: ./car.yaml

car.yaml
    parameters:
      - name: carId
        in: path
        required: true
        description: vin number
        type: string
      - name: companyId
        in: path
        required: true
        description: Id of the company that makes the car
        type: string
    get:
     summary: Returns the car
     description: Returns the car
     responses:
        200:
          description: Successful response OK

**Comment by mohsen1** [comment]
Where `car.yaml` is hosted? To resolve the document those URLs should be reachable

**Comment by Tasneem1434** [comment]
its a file stored locally on my machine..

**Comment by mohsen1** [comment]
You should serve those files where editor is being server *or* use absolute URLs to your server

**Comment by Tasneemb** [comment]
Could you please give me an example of both . My swagger-editor is opened
with url http://localhost:8080/#/...but installed in a folder on my C:
called swagger-editor... I run npm start from c:\swagger-editor.

Thank you for your reponses

**Comment by mohsen1** [comment]
So if your `index.yaml` is a sibling of `index.html` of Swagger Editor in it should be able to resolve it.

**Comment by mohsen1** [comment]
We found a bug. Relative paths do not work for now due to this issue:

https://github.com/whitlockjc/path-loader/issues/5

**Comment by Tasneemb** [comment]
so should I be using only absolute URL's? If yes, please provide an
example...

Thank you

**Comment by mohsen1** [comment]
Example absolute path:

```
definitions:
  $ref: http://localhost:8080/files/definitions/index.yaml
```

**Comment by Tasneem1434** [comment]
Hi

I changed all my $ref to have absolute paths as above. Swagger-editor "dist" folder has been copied to Apache Tomcat webapps folder. Not getting any errors anymore, but when I try and generate html from the menu, it gives error "Oops. Something bad has happened...." Seems the problem is  with $ref, because when I put the actual yaml in the places where I have $ref, the HTML gets generated fine....

Please assist.

Thanking you

**Comment by mohsen1** [comment]
Fixed in c80708b

**Comment by Tasneem1434** [comment]
Hi. I replaced swagger-editor with the latest version (2.9.8). Now when I try to Generate Client- >Dynamic HTML or any other menu item, it does nothing. Please note that this is a yaml file where there is use of $ref:..But if I have to open up an example like default.yaml, the Generate Client->Dynamic HTML works perfectly. please advise
